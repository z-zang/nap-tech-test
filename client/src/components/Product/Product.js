import React, {useState, useEffect} from 'react'

function Product({pid, name, price, designer, img}) {
  return (
    <div id={pid} className="product">
      <p>{designer}</p>
      <img src={img} alt={name} className="prodImg"/>
      <p className="productText">{name}</p>
      <p>Â£{price}</p>
    </div>
  )
}

export default Product