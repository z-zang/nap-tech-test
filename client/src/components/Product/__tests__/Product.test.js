import React from 'react';
import Product from '../Product';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, configure } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('Product', () => {
  let props;
  beforeAll(() => {
    props = {
      pid: '504815', 
      name:'Cutout stretch-jersey dress', 
      price: '1270', 
      designer: 'Donna Karan', 
      img: '//cache.net-a-porter.com/images/products/504815/504815_ou_sl.jpg'
    };
  })

  it('should match snapshot', async () => {
    const wrapper = mount(<MemoryRouter><Product {...props} /></MemoryRouter>);
    expect(wrapper.find(Product)).toMatchSnapshot();

});
  

})

