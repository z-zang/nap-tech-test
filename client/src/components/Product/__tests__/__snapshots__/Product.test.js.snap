// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
<Product
  designer="Donna Karan"
  img="//cache.net-a-porter.com/images/products/504815/504815_ou_sl.jpg"
  name="Cutout stretch-jersey dress"
  pid="504815"
  price="1270"
>
  <div
    className="product"
  >
    <Link
      className="router-link"
      to="/product/504815"
    >
      <a
        className="router-link"
        href="/product/504815"
        onClick={[Function]}
      >
        <div
          className="product-container"
          id="504815"
        >
          <p>
            Donna Karan
          </p>
          <img
            alt="Cutout stretch-jersey dress"
            className="prodImg"
            src="//cache.net-a-porter.com/images/products/504815/504815_ou_sl.jpg"
          />
          <p
            className="productText"
          >
            Cutout stretch-jersey dress
          </p>
          <p>
            Â£
            1270
          </p>
        </div>
      </a>
    </Link>
  </div>
</Product>
`;
